handlers/handler.go

```diff
@@ -36,26 +36,37 @@ const dayStarFormat = "%3v"
 
 func CommandHandler(session *discordgo.Session, message *discordgo.MessageCreate) {
 	msgContent := strings.TrimSpace(message.Content)
-	//TODO: refactor this with slash commands and have year be an argument, current year as default
-
-	if strings.HasPrefix(msgContent, "/aoc2020") {
-		parse(session, message, 2020)
-	} else if strings.HasPrefix(msgContent, "/aoc2021") {
-		parse(session, message, 2021)
-	} else if strings.HasPrefix(msgContent, "/aoc2022") {
-		parse(session, message, 2022)
-	} else if strings.HasPrefix(msgContent, "/aoc") {
-		if time.Now().Month() < 12 { //if it's not yet december, just take last years leaderboard
-			parse(session, message, time.Now().Year()-1)
+
+	if strings.HasPrefix(msgContent, "/aoc-info") {
+		infotext := "Advent Of Code: https://adventofcode.com/2022/about \n" +
+			"Join our leaderboard from https://adventofcode.com/2022/leaderboard/private with this join code: `784176-b767a0f2`.\n" +
+			"Use `/aoc year` command to see the current leaderboard."
+		session.ChannelMessageSend(message.ChannelID, infotext)
+		return
+	}
+
+	if strings.HasPrefix(msgContent, "/aoc") {
+		_, after, found := strings.Cut(msgContent, " ")
+		if found {
+			if intVar, err := strconv.ParseInt(after, 10, 32); err == nil {
+				if intVar < 2015 || intVar > int64(time.Now().Year()+1) {
+					return
+				}
+				parse(session, message, int((intVar)))
+			}
```

Comment:
Please put both if conditions in one and on the else branch call the function from the else branch below which calls current year or year before